src/Illuminate/Support/Testing/Fakes/BatchRepositoryFake.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Illuminate\Support\Testing\Fakes;
+
+use Carbon\CarbonImmutable;
+use Closure;
+use Illuminate\Bus\Batch;
+use Illuminate\Bus\BatchRepository;
+use Illuminate\Bus\PendingBatch;
+use Illuminate\Support\Facades\Facade;
+use Illuminate\Support\Str;
+
+class BatchRepositoryFake implements BatchRepository
+{
+    /**
+     * Retrieve information about an existing batch.
+     *
+     * @param  string  $batchId
+     * @return \Illuminate\Bus\Batch|null
+     */
+    public function find(string $batchId)
+    {
+    }
+
+    /**
+     * Store a new pending batch.
+     *
+     * @param  \Illuminate\Bus\PendingBatch  $batch
+     * @return \Illuminate\Bus\Batch
+     */
+    public function store(PendingBatch $batch)
+    {
+        return new Batch(
+            new QueueFake(Facade::getFacadeApplication()),
+            $this,
+            (string) Str::orderedUuid(),
+            count($batch->jobs),
+            count($batch->jobs),
+            0,
+            $batch->options,
+            CarbonImmutable::now(),
+            null,
+            null
+        );
+    }
+
+    /**
+     * Increment the total number of jobs within the batch.
+     *
+     * @param  string  $batchId
+     * @param  int  $amount
+     * @return void
+     */
+    public function incrementTotalJobs(string $batchId, int $amount)
+    {
+    }
+
+    /**
+     * Decrement the total number of pending jobs for the batch.
+     *
+     * @param  string  $batchId
+     * @return int|null
+     */
+    public function decrementPendingJobs(string $batchId)
```

Comment:
This is `BatchRepositoryFake implements BatchRepository` but the method on the interface is `public function decrementPendingJobs(string $batchId, string $jobId);`, as such this doesn't work due to:
> `PHP Fatal error:  Declaration of Illuminate\Support\Testing\Fakes\BatchRepositoryFake::decrementPendingJobs(string $batchId) must be compatible with Illuminate\Bus\BatchRepository::decrementPendingJobs(string $batchId, string $jobId) in …/src/Illuminate/Support/Testing/Fakes/BatchRepositoryFake.php on line …`